lib/reverse_sentence.rb

```diff
@@ -1,6 +1,98 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string.
+#                                 (more line O(3n)
+# Space complexity: O(1) - in place
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence.nil? || my_sentence.length == 0
+
+  # reversing the whole string
+  i = 0
+  while i < my_sentence.length / 2
+    temp = my_sentence[i]
+    my_sentence[i] = my_sentence[my_sentence.length - 1 - i]
+    my_sentence[my_sentence.length - 1 - i] = temp
+    i += 1
+  end
```

Comment:
Notice that you're reversing things twice here.  Could you abstract that job into a helper function?  That function might look like:  `reverse_substring(string, start, end)`  Then you could use it in each task.